luma/core/framebuffer.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014-17 Richard Hull and contributors
+# See LICENSE.rst for details.
+
+"""
+Different implementation strategies for framebuffering
+"""
+
+from PIL import Image, ImageChops
+
+
+class diff_to_previous(object):
+    """
+    Compare the current frame to the previous frame and tries to calculate the
+    differences: this will either be ``None`` for a perfect match or some
+    bounding box describing the areas that are different, upto the size of the
+    entire image.
+
+    The image data for the difference is then be passed to a device for
+    rendering just those small changes. This can be very quick for small screen
+    updates, but suffers from variable render times, depending on the changes
+    applied. The :py:class`luma.core.sprite_system.framerate_regulator` may be
+    used to counteract this behviour however.
+
+    :param device: the target device, used to determine the initial 'previous'
+        image.
+    :type device: luma.core.device.device
+    """
+    def __init__(self, device):
+        self.image = Image.new(device.mode, device.size, "white")
+
+    def update(self, image):
+        """
+        Calculates the difference from the previous image, setting ``bbox`` and
+        ``image`` attributes, and priming :py:func:`getdata`.
+
+        :param image: An image to render
+        :type image: PIL.Image.Image
+        """
+        self.bbox = ImageChops.difference(self.image, image).getbbox()
+        if self.bbox is not None:
+            self.image = image.copy()
+
+    def getdata(self):
+        """
+        A sequence of pixel data relating to the changes that occurred
+        since the last time :py:func:`update` was last called.
+
+        :returns: A sequence of pixels
```

Comment:
Strictly speaking it is not an image, but an iterable of (a tuple of RGB bands or a single intensity value):

```
 $ python
Python 2.7.13 (default, Dec 19 2016, 01:44:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from PIL import Image
>>> x = Image.new("1", (40,40))
>>> x
<PIL.Image.Image image mode=1 size=40x40 at 0x10167CB50>
>>> x.getdata()
<ImagingCore object at 0x101567190>
```